resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/ReactorCircuitBreakerAspect.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Set;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Component;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.utils.CircuitBreakerUtils;
+import io.github.resilience4j.reactor.circuitbreaker.operator.CircuitBreakerOperator;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * the Reactor breaker logic support for the spring AOP
+ * Conditional on Reactor class existence on spring class loader
+ */
+
+@Component
+@Conditional(value = {InjectReactorAspect.class})
+class ReactorCircuitBreakerAspect implements CircuitBreakerAspectExt {
+
+	private static final Logger logger = LoggerFactory.getLogger(ReactorCircuitBreakerAspect.class);
+
+	public ReactorCircuitBreakerAspect() {
+	}
+
+	@SafeVarargs
+	private static <T> Set<T> newHashSet(T... objs) {
+		Set<T> set = new HashSet<>();
+		Collections.addAll(set, objs);
+		return Collections.unmodifiableSet(set);
+	}
+
+	/**
+	 * @param returnType the AOP method return type class
+	 * @return boolean if the method has Reactor return type
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean matchReturnType(Class returnType) {
+		return (Flux.class.isAssignableFrom(returnType)) || (Mono.class.isAssignableFrom(returnType));
+	}
+
+	/**
+	 * handle the Spring web flux (Flux /Mono) return types AOP based into reactor circuit-breaker
+	 * See {@link io.github.resilience4j.reactor.circuitbreaker.operator.CircuitBreakerOperator} for details.
+	 *
+	 * @param proceedingJoinPoint Spring AOP proceedingJoinPoint
+	 * @param circuitBreaker      the configured circuitBreaker
+	 * @param methodName          the method name
+	 * @return the result object
+	 * @throws Throwable exception in case of faulty flow
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public Object handle(ProceedingJoinPoint proceedingJoinPoint, CircuitBreaker circuitBreaker, String methodName) throws Throwable {
+		CircuitBreakerUtils.isCallPermitted(circuitBreaker);
+		long start = System.nanoTime();
+		try {
+			Object returnValue = proceedingJoinPoint.proceed();
+			if (Flux.class.isAssignableFrom(returnValue.getClass())) {
+				Flux fluxReturnValue = (Flux) returnValue;
+				return fluxReturnValue.transform(io.github.resilience4j.reactor.circuitbreaker.operator.CircuitBreakerOperator.of(circuitBreaker));
+			} else if (Mono.class.isAssignableFrom(returnValue.getClass())) {
+				Mono monoReturnValue = (Mono) returnValue;
+				return monoReturnValue.transform(CircuitBreakerOperator.of(circuitBreaker));
+			} else {
+				throw new IllegalArgumentException("Not Supported type for the circuit breaker in web flux :" + returnValue.getClass().getName());
+
+			}
+		} catch (Throwable throwable) {
+			long durationInNanos = System.nanoTime() - start;
```

Comment:
I think we can remove it. We don't have to hande the throwable. This exception signals a development bug and wrong usage of Spring Reactor. Must not be captured by a CircuitBreaker.

Reply:
I think we should not capture development bugs, otherwise the CircuitBreaker opens, even if the backend is fine.